docs/content/manual/pre-release/ha/ha-volume-migration.md

```diff
@@ -2,30 +2,400 @@
+
+Confirmation succeeds. Then, the volume detaches from and reattaches to the old node.
+
+```bash
+k delete --wait=false -f va-2.yaml && sleep 0.5 && kl delete pod instance-manager-699da83c0e9d22726e667344227e096b
+# OR
+k delete --wait=false -f va-2.yaml && sleep 1 && kl delete pod instance-manager-699da83c0e9d22726e667344227e096b
+```
+
+- "Rolling back migration"
+- "...selected to detach from <old>"
+- "...selected to attach to <old>"
+- Same engine and replicas (rolled back)
+
+### 3. Single replica crash
+
+Crash the replica on a node that is neither the old or migration node by cordoning the node and killing its
+instance-manager pod.
+
+#### 3.1 Degraded before migration and confirmation
+
+Migration starts while the volume is degraded. Confirmation succeeds.
+
+```bash
+k cordon eweber-v126-worker-9c1451b4-rw5hf
+kl delete pod instance-manager-6852914a55e4566d3ddea43529df22e0
+k delete -f va-1.yaml
+```
+
+- "Confirming migration"
+- No detachment
+- New engine and replicas
+
+#### 3.2 Degraded before migration and rollback
+
+Migration starts while the volume is degraded. Rollback succeeds.
+
+```bash
+k cordon eweber-v126-worker-9c1451b4-rw5hf
+kl delete pod instance-manager-6852914a55e4566d3ddea43529df22e0
+
+k apply -f va-1.yaml
+k apply -f va-2.yaml
+k delete -f va-2.yaml
+```
+
+- "Rolling back migration"
+- No detachment
+- Same engine and replicas
+
+#### 3.3 Degraded between migration start and confirmation
+
+Confirmation succeeds.
+
+```bash
+k apply -f va-1.yaml
+k apply -f va-2.yaml
+
+k cordon eweber-v126-worker-9c1451b4-rw5hf
+kl delete pod instance-manager-6852914a55e4566d3ddea43529df22e0
+
+k delete -f va-1.yaml
+```
+
+- "Confirming migration"
+- New engine and replicas
+
+#### 3.4 Degraded between migration start and rollback
+
+Rollback succeeds.
+
+```bash
+k apply -f va-1.yaml
+k apply -f va-2.yaml
+
+k cordon eweber-v126-worker-9c1451b4-rw5hf
+kl delete pod instance-manager-6852914a55e4566d3ddea43529df22e0
+
+k delete -f va-2.yaml
+```
+
+- Rolling back migration
+- Same engine and replicas
+
+### 4. Attempt to attach to three nodes
+
+The third attachment fails.
+
+```bash
+kl apply -f va-1.yaml
+kl apply -f va-2.yaml
+kl apply -f va-3.yaml
+```
+
+- test-va-1 attached
+- test-va-2 attached
+- test-va-3 not attached
+- "...cannot attach migratable volume to more than two nodes..."
+
+### 5. New engine node down
+
+1. Hard shut down the node running the migration engine.
+2. Wait until Kubernetes recognizes the node is down. (This is IMPORTANT! Otherwise, it is a different test case.)
+3. Attempt a confirmation or rollback.
+
+#### 5.1 Confirmation
+
+The volume is allowed to detach from the old node (special logic in code). It attempts to attach to cleanly attach to
+the migration node, but is stuck until it comes back.
+
+- "Waiting to confirm migration until migration engine is ready"
+- "Detaching volume for attempted migration to down node"
+- "...selected to attach to <new>"
+- Same engine and replicas
```

Comment:
Is it possible to provide more information on how to check "Same engine and replicas"? Sorry but I'm a bit confused..

Reply:
I added https://github.com/longhorn/longhorn-tests/pull/1948/files#diff-b892c032654f9b2b7dcd1d87cdd7c8333ab3eb449f5a71213990f3bb8ebdc6ffR86-R91 for clarity.

In practice, it means if you check the engine and replica names before the scenario and then after, they may be the same (e.g. for a migration failure or a successful migration rollback) or different (e.g. for a successful migration confirmation). In the middle of the migration, we always have two engines with different names and two sets of three replicas with different names.

Please let me know if you think I should add additional context!